### PR TITLE
Reland "Fix CheckExchangeMessages crashes when malicious packet is sent"

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
This is reland of https://github.com/project-chip/connectedhomeip/pull/6120 which was reverted because of regression in PASE.

The root cause of the regression is PASE is moved from transport layer to exchange, and packets exchanged during PASE do not have Source Node ID, so the following check in MatchExchange is no more valid.  

  // AND The message must come with a source Node ID.
  && (packetHeader.GetSourceNodeId().HasValue())

 #### Summary of Changes
Fix CheckExchangeMessages crashes when malicious packet is sent

 Fixes ##4635

 #### Test
 I don't have ESP32, so I used python device controller and lighting example to verify this change. I expect the PASE logic is same cross different platforms.
 
 chip-device-ctrl > connect -ip 192.168.86.157 20202021 12344321
Device is assigned with nodeid = 12344321
CHIP:BLE: Assigned local session key ID 0
CHIP:BLE: Sent PBKDF param request
CHIP:BLE: Received PBKDF param response
CHIP:BLE: Sent spake2p msg1
CHIP:BLE: Received spake2p msg2
CHIP:BLE: Peer assigned session key ID 0
CHIP:BLE: Sent spake2p msg3
CHIP:IN: New pairing for device 0x0000000000bc5c01, key 0!!
CHIP:CTL: Remote device completed SPAKE2+ handshake

CHIP:CTL: AsyncSetKeyValue: PairedDevicebc5c01=EAABEbyplB3xp6cuEWz2SFVrAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAFcvAAAAAAAMTkyLjE2OC44Ni4xNTcAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFkrAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAA
CHIP:CTL: AsyncSetKeyValue: ListPairedDevices0=AVy8AAAAAAA=
CHIP:CTL: AsyncSetKeyValue: StartKeyID=1
Secure Session to Device Established
Device temporary node id (**this does not match spec**): 12344321
chip-device-ctrl >

### CODE REVIEW

File: src/messaging/tests/TestExchangeMgr.cpp [hunk 1]
```diff
@@ -153,9 +153,9 @@ void CheckExchangeMessages(nlTestSuite * inSuite, void * inContext)
     NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
 
     // send a malicious packet
-    // TODO: https://github.com/project-chip/connectedhomeip/issues/4635
-    // ec1->SendMessage(0x0001, 0x0002, System::PacketBufferHandle::New(System::PacketBuffer::kMaxSize),
-    //                 SendFlags(Messaging::SendMessageFlags::kNone));
+    ec1->SendMessage(Protocols::Id(VendorId::Common, 0x0001), 0x0002,
```
Comment (woody-apple) [inline src/messaging/tests/TestExchangeMgr.cpp:156]: Should these be constants?

Comment (yufengwangca) [reply src/messaging/tests/TestExchangeMgr.cpp]: Acked

**Review by andy31415 — APPROVED** [review]

**Comment by yufengwangca** [comment]
@woody-apple ?

**Review by vivien-apple — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst @msandstedt  ?

**Review by saurabhst — APPROVED** [review]

**Pull request merged**